examples/ecs/hotpatching_systems.rs

```diff
@@ -0,0 +1,77 @@
+//! This example demonstrates how to hotpatch systems.
+//!
+//! It needs to be run with the dioxus CLI:
+//! ```sh
+//! dx serve --hot-patch --example hotpatching_systems --features hotpatching
+//! ```
+//!
+//! You can change the text in the `update_text` system, or the color in the
+//! `on_click` system, and those changes will be hotpatched into the running
+//! application.
+
+use std::time::Duration;
+
+use bevy::{color::palettes, prelude::*};
+
+fn main() {
+    let (sender, receiver) = crossbeam_channel::unbounded::<()>();
+
+    std::thread::spawn(move || {
+        while receiver.recv().is_ok() {
+            let start = bevy::platform::time::Instant::now();
+            // You can also make any part outside of a system hot patchable by wrapping it
+            // In this part, only the duration is hot patchable:
+            let duration = bevy::dev_tools::hotpatch::call(|| Duration::from_secs(2));
+            std::thread::sleep(duration);
+            info!("done after {:?}", start.elapsed());
+        }
+    });
```

Comment:
I'd move this into a separate example or make sure it is much lower in the file. Right now, this is the first code users see when they look at hot-patching, which will give them the idea that they need to do weird thread stuff.

Reply:
tried to improve the example while keeping everything in one